lib/mix/test/mix/dep/lock_test.exs

```diff
@@ -25,4 +25,20 @@ defmodule Mix.Dep.LockTest do
       refute File.regular? "_build/dev/lib/sample/.compile.lock"
     end
   end
+
+  test "raises a proper error for merge conflicts", context do
+    in_tmp context.test, fn ->
+      File.write "mix.lock", ~S"""
+        <<<<<<< HEAD
+        %{"foo": {:hex, :foo, "0.1.0"},
+        =======
+        %{"bar": {:hex, :bar, "0.1.0"},
+        >>>>>>> foobar
+        %{"baz": {:hex, :baz, "0.1.0"}}
+        """
+      assert_raise Mix.Error, ~r(Your mix.lock contains merge conflicts), fn ->
```

Comment:
I think the convention is to use `~r//` when there's no need to escape `/`s. Also, `mix\.lock` :)